sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -11,52 +11,58 @@ import com.horizen.certnative.BackwardTransfer
 import com.horizen.transaction.exception.TransactionSemanticValidityException
 import com.horizen.transaction.{CertificateKeyRotationTransaction, MC2SCAggregatedTransaction, OpenStakeTransaction, SidechainTransaction}
 import com.horizen.utils.{BlockFeeInfo, ByteArrayWrapper, BytesUtils, FeePaymentsUtils, MerkleTree, TimeToEpochUtils, WithdrawalEpochInfo, WithdrawalEpochUtils}
-import sparkz.crypto.hash.Blake2b256
-import sparkz.util.{ModifierId, SparkzLogging, bytesToId}
+import com.horizen.validation.crosschain.CrossChainValidator
+import com.horizen.validation.crosschain.receiver.CrossChainRedeemMessageValidator
+import com.horizen.validation.crosschain.sender.CrossChainMessageValidator
 import sparkz.core._
 import sparkz.core.transaction.state._
+import sparkz.crypto.hash.Blake2b256
+import sparkz.util.{ModifierId, SparkzLogging, bytesToId}
+
 import java.io.File
 import java.math.{BigDecimal, MathContext}
 import java.util
 import java.util.{ArrayList => JArrayList, HashMap => JHashMap, Optional => JOptional}
-
-import com.horizen.sc2sc.{CrossChainMessage, CrossChainMessageHash, CrossChainMessageImpl, Sc2ScConfigurator}
-
 import scala.collection.JavaConverters._
 import scala.collection.mutable.ListBuffer
 import scala.util.{Failure, Success, Try}
 
 
-class SidechainState private[horizen] (stateStorage: SidechainStateStorage,
-                                       forgerBoxStorage: SidechainStateForgerBoxStorage,
-                                       utxoMerkleTreeProvider: SidechainStateUtxoMerkleTreeProvider,
-                                       val params: NetworkParams,
-                                       val sc2scConfig : Sc2ScConfigurator,
-                                       override val version: VersionTag,
-                                       val applicationState: ApplicationState)
+class SidechainState private[horizen](stateStorage: SidechainStateStorage,
+                                      forgerBoxStorage: SidechainStateForgerBoxStorage,
+                                      utxoMerkleTreeProvider: SidechainStateUtxoMerkleTreeProvider,
+                                      val params: NetworkParams,
+                                      val sc2scConfig: Sc2ScConfigurator,
+                                      sidechainSettings: SidechainSettings,
+                                      override val version: VersionTag,
+                                      val applicationState: ApplicationState)
   extends AbstractState[SidechainTypes#SCBT, SidechainBlockHeader, SidechainBlock, SidechainState]
     with TransactionValidation[SidechainTypes#SCBT]
     with ModifierValidation[SidechainBlock]
     with SidechainTypes
     with NodeState
     with SparkzLogging
     with UtxoMerkleTreeView
-    with NetworkParamsUtils
-{
-
-
+    with NetworkParamsUtils {
   override type NVCT = SidechainState
+  type PMOD = SidechainBlock
+
+  protected val crossChainValidators: Seq[CrossChainValidator[PMOD]] = Seq(
```

Comment:
The idea is to pass the custom validators from the Application side. I think we need to add an additional injected object inside the **SidechainAppModule** like "Sc2ScMessageValidators" which is a `Seq[CrossChainValidator]` and pass it to the **SidechainState** constructor.